sigstore-public-deployment-spec.md

```diff
@@ -0,0 +1,172 @@
+
+##### 2.2.1.1 Requirements on Identity Providers
+
+[https://github.com/sigstore/fulcio/blob/main/docs/new-idp-requirements.md](https://github.com/sigstore/fulcio/blob/main/docs/new-idp-requirements.md)
+
+##### 2.2.1.2 Current Identity Providers
+
+Sigstore runs a federated OIDC identity provider, [Dex](https://dexidp.io/). Users authenticate to their preferred identity provider and Dex creates an OIDC token with claims from the original OIDC token. Fulcio also supports OIDC tokens from additional configured issuers.
+
+The Fulcio implementation allows deployers to configure which OIDC IdPs to accept. The public good instance accepts the following:
+
+**User Authentication**
+
+Dex:
+
+* GitHub  
+* Microsoft  
+* Google
+
+**Workflow Authentication**
+
+* GitHub  
+* GitLab  
+* BuildKite
+
+See the [Fulcio OIDC documentation](https://github.com/sigstore/fulcio/blob/main/docs/oidc.md) for additional details.
+
+#### 2.2.2 SPIFFE
+
+[Secure Production Identity Framework for Everyone](https://spiffe.io/) (SPIFFE) uses X.509 certificates to provide identity. SPIFFE-based OIDC providers use a SPIFFE ID as the URI subject alternative name of the certificate, scoped to a domain.
+
+## 3. Rekor
+
+Rekor implements a transparency service. There is a public good deployment of Rekor run by the [OpenSSF](https://openssf.org/) and contributing organizations at [https://rekor.sigstore.dev/](https://rekor.sigstore.dev/).
+
+### 3.1 Pluggable Types
+
+The transparency service has what is termed a ‘pluggable type’ system. A pluggable type, is a custom schema for entries stored in the transparency log. Schemas can be in multiple formats (json|yaml|xml).
+
+The current list of supported types can be found in the [Rekor project](https://github.com/sigstore/rekor/tree/main/pkg/types). Information about adding new pluggable types can be found in the [Rekor documentation.](https://docs.sigstore.dev/docs/logging/pluggable-types/)
+
+See the transparency service ([Spec: Rekor](https://docs.google.com/document/u/0/d/1NQUBSL9R64_vPxUEgVKGb0p81_7BVZ7PQuI078WFn-g/edit)) document for additional information.
+
+### 3.2 Transparency Log
+
+Rekor is backed by a transparency log, inspired by the one in Certificate Transparency ([RFC 6962](https://datatracker.ietf.org/doc/rfc6962/)). The primary deviations are described in [Spec: Rekor](https://docs.google.com/document/u/0/d/1NQUBSL9R64_vPxUEgVKGb0p81_7BVZ7PQuI078WFn-g/edit); however, like RFC 6962, the transparency service specification leaves specific parameter choices to implementers.
+
+* Base URL: [https://rekor.sigstore.dev/](https://rekor.sigstore.dev/)   
+* Hash Algorithm: SHA-256 ([RFC 6234](https://datatracker.ietf.org/doc/rfc6234/); OID 2.16.840.1.101.3.4.2.1)  
+* Signature Algorithm:  ECDSA (NIST P-256).  
+* Public Key: change over time  
+* Log ID: need an OID  
+* Maximum Merge Delay: Rekor only returns after the merge is complete  
```

Comment:
We should flesh this line out, as this is a major differentiator from CT. Rekor returns a log index to the caller which can only be obtained since Trillian will only give an index once the entry is merged.
This is different with the tile implementations, which return indices but still need to wait for inclusion (iirc)

Reply:
Updated this using info from the Rekor v2 doc.